File: Etiquette/lib/Providers/DB/update_ticket_owner.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:Etiquette/Models/serverset.dart';

Future<Map<String, dynamic>> updateTicketOwner(String owner, String token_id) async {
  try {
    const url = "$SERVER_IP/ticket/updateTicketOwner";
    final res = await http.post(Uri.parse(url), body: {
      "owner": owner,
      "token_id": token_id,
    });
    Map<String, dynamic> data = json.decode(res.body);
    return data;
  } catch (ex) {
    Map<String, dynamic> data = {
      "statusCode": 400,
      "msg": ex.toString()
    };
    return data;
  }
}

File: Backend/controller/ticket/updateTicketOwner.js
const Ticket = require('../../model/ticket');

const updateTicketOwner = async function (req, res) {
    try {
        Ticket.setUpdateTicketOwner(req.body.owner, req.body.token_id, function (err, row) {
            if (!err) {
                res.status(200);
                res.json({
                    statusCode: 200
                });
            } else {
                res.status(401)
                res.json({
                    statusCode: 401,
                    msg: "서버와의 상태가 원활하지 않습니다. 잠시 후 다시 시도해주세요."
                });
                console.log(`updateTicketOwner: ${err}`);
            }
        })
    } catch (e) {
        console.error(e);
    }
}

module.exports = updateTicketOwner;